src/test/java/org/takes/tk/TkCORSTest.java

```diff
@@ -0,0 +1,132 @@
+
+/**
+ * Test case for {@link TkCORS}.
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ */
+public class TkCORSTest {
+
+    /**
+     * Execute TkCORS don't sending any origin on the request.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public final void withoutDomain() throws IOException {
+        final Take take = Mockito.mock(Take.class);
+        Mockito.doReturn(new RsText()).when(take)
+            .act(Mockito.any(Request.class));
+        final Take cors = new TkCORS(take);
+        final RsPrint response = new RsPrint(cors.act(new RqFake()));
+        final String head = response.printHead();
+        this.httpStatusOk(head);
+    }
+
+    /**
+     * Execute TkCORS sending a origin that is on the allowed domains to
+     * perform the action.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public final void withCorrectDomain() throws IOException {
+        final Take take = Mockito.mock(Take.class);
+        Mockito.doReturn(new RsText()).when(take)
+        .act(Mockito.any(Request.class));
+        final Take cors = new TkCORS(
+            take,
+            "http://teamed.io",
+            "http://example.com"
+        );
+        final Set<String> headers = new HashSet<String>();
+        headers.add("Origin: http://teamed.io");
+        final Request req = new RqWithHeaders(new RqFake(), headers);
+        final RsPrint response = new RsPrint(cors.act(req));
+        final String head = response.printHead();
+        this.httpStatusOk(head);
+    }
+
+    /**
+     * Execute TkCORS sending a origin that is on the allowed domains to
+     * perform the action. TODO
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public final void withWrongDomain() throws IOException {
+        final Take take = Mockito.mock(Take.class);
+        Mockito.doReturn(new RsText()).when(take)
+        .act(Mockito.any(Request.class));
+        final Take cors = new TkCORS(
+            take,
+            "http://www.teamed.io",
+            "http://sample.com"
+        );
+        final Set<String> headers = new HashSet<String>();
+        headers.add("Origin: http://wrong.teamed.io");
+        final Request req = new RqWithHeaders(new RqFake(), headers);
+        final RsPrint response = new RsPrint(cors.act(req));
+        final String head = response.printHead();
+        MatcherAssert.assertThat(
+            "It was expected a 403 HTTP status.",
+            head,
+            Matchers.containsString("HTTP/1.1 403")
+        );
+        MatcherAssert.assertThat(
+            "Wrong value on Access Control Allow Credentias param.",
+            head,
+            Matchers.containsString("Access-Control-Allow-Credentials: false")
+        );
+    }
+
+    /**
+     * Checks if the given head contains text that identify it with HTTP 200
+     * status.
+     * @param head HTTP response head.
+     */
+    @Ignore
```

Comment:
@antonini Why is this method ignored? It won't be picked up by JUnit, no need for the annotation.